整合HttpClient，并添加登录和获取题目内容相关代码

KamenRiderKuuga opened the pull request.

21kl committed:
Merge branch 'main' of C:\Users\cb151\Desktop\bb\qmoj-core with conflicts.

KamenRiderKuuga committed:
整合HttpClient，并添加登录和获取题目内容相关代码

timeaftertime commented on qmoj/src/main/java/team/akina/qmoj/constants/Constants.java:
```diff
@@ -0,0 +1,24 @@
+    public static final String SIGN_IN_URL = "https://leetcode-cn.com/accounts/login/";
+
+    // 请求时默认携带的User-Agent
+    public static final String USER_AGENT = "Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 5.1)";
```
这个参数是必须的吗

timeaftertime commented on qmoj/src/main/java/team/akina/qmoj/controller/HelloController.java:
```diff
@@ -1,14 +1,29 @@
+    private LeetCodeHelper leetCodeHelper;
+
+    @Autowired
+    public HelloController(LeetCodeHelper leetCodeHelper) {
```
这里为什么不直接在字段上添加 @Autowired

timeaftertime commented on qmoj/src/main/java/team/akina/qmoj/controller/HelloController.java:
```diff
@@ -1,14 +1,29 @@
+    * @return 题目详情
+    */
+    @RequestMapping("/helloqmoj/{slug}")
+    public String hello(@PathVariable String slug) throws Exception {
```
Controller 层不应该抛出异常吧，虽然只是个测试方法

timeaftertime commented on qmoj/src/main/java/team/akina/qmoj/controller/HelloController.java:
```diff
@@ -1,14 +1,29 @@
+    */
+    @RequestMapping("/helloqmoj/{slug}")
+    public String hello(@PathVariable String slug) throws Exception {
+        return leetCodeHelper.getProblemDetailBySlug(slug);
```
返回给前端的格式是不是需要固定下呢

timeaftertime commented on qmoj/src/main/java/team/akina/qmoj/utils/LeetCodeHelper.java:
```diff
@@ -0,0 +1,84 @@
+     *
+     * @return 登录结果
+     */
+    public boolean loginToLeetCode() throws Exception {
```
好像到处都有 throws Exception ，可能有必要讨论一下

timeaftertime commented on qmoj/src/main/java/team/akina/qmoj/utils/LeetCodeHelper.java:
```diff
@@ -0,0 +1,84 @@
+        headers.put("Referer", Constants.SIGN_IN_URL);
+
+        httpRequestHelper.doPost(Constants.SIGN_IN_URL, formContent, headers);
+        return httpCookieStore.getCookies().size() >= 3;
```
这个判断的含义好像不太清晰

timeaftertime commented on qmoj/src/main/java/team/akina/qmoj/utils/StringUtil.java:
```diff
@@ -0,0 +1,68 @@
+     * @param unicode unicode字符串
+     * @return 转换完成的字符串
+     */
+    public static String decodeUnicode(final String unicode) {
```
这里转换逻辑似乎没必要这么复杂，可以使用 Integer.parseInt 处理 Unicode 值

timeaftertime commented on qmoj/src/main/java/team/akina/qmoj/utils/http/HttpClientUtil.java:
```diff
@@ -0,0 +1,114 @@
+     * @return httpClient实例
+     */
+    @Bean
+    public CloseableHttpClient getCloseableHttpClient(@Qualifier("httpClientBuilder") HttpClientBuilder httpClientBuilder) {
```
这个方法的目的是什么？为什么不直接调用呢

timeaftertime commented on qmoj/src/main/java/team/akina/qmoj/utils/http/HttpRequestHelper.java:
```diff
@@ -0,0 +1,166 @@
+    }
+
+    /**
+     * 无参get请求，如果状态码为200，则返回body，否则返回null
```
很多方法都可能返回 null ，是否合理可能需要讨论下

21kl committed:
简单整合mybatis，样例demo

21kl committed:
Merge branch 'dev' of github.com:qms-dev/qmoj-core into dev

# Conflicts:
#	qmoj/src/main/java/team/akina/qmoj/controller/HelloController.java
#	qmoj/src/main/resources/application.yml

21kl committed:
简单整合mybatis，样例demo

KamenRiderKuuga committed:
删除idea相关文件

21kl committed:
entity,mapper去除ignore

21kl committed:
entity,mapper去除ignore

21kl committed:
entity,mapper去除ignore

21kl committed:
Merge branch 'dev' of github.com:qms-dev/qmoj-core into dev

# Conflicts:
#	.idea/workspace.xml

gk008 committed:
对获取题目列表的json结果做了解析

gk008 committed:
提交实体类！

commit之前没有提交的内容

gk008 committed:
根据畅爷的comment修改一些小地方

gk008 committed:
修复一处bug

只需请求一次LeetCode拿到list即可，- -，实在过于粗糙。

gk008 committed:
做了题目内容的解析，代码非常难看，请大佬们疯狂comment

非常大量的重复代码

KamenRiderKuuga committed:
添加自定义异常，整理第一次pr的代码，添加了两点todo

KamenRiderKuuga committed:
调整上次提交的函数的功能，添加一些todo项

KamenRiderKuuga committed:
调整代码中的"Problem"为"Question"，调整自定义异常类，使其可以对参数进行记录

KamenRiderKuuga committed:
Merge pull request #2 from qms-dev/dev-hanabi

调整代码中的"Problem"为"Question"，调整自定义异常类，使其可以对参数进行记录

gk008 committed:
Revert "做了题目内容的解析，代码非常难看，请大佬们疯狂comment"

This reverts commit 38dcf07fcc570e8061c43d434f138616a41771c8.

gk008 committed:
Revert "修复一处bug"

This reverts commit 2914c993c70709802920daca338185add37792dc.

gk008 committed:
Revert "根据畅爷的comment修改一些小地方"

This reverts commit 8a50b3d19d508ed65d9dc1208ebb247c29ae22e6.

gk008 committed:
Merge branch 'dev' of https://github.com/qms-dev/qmoj-core into dev

gk008 committed:
提交到GK分支

KamenRiderKuuga committed:
Merge pull request #3 from qms-dev/dev-GK

合并GK之前未提交的代码

KamenRiderKuuga committed:
增加之前没有考虑到的字段question_tags，用于记录题目标签

KamenRiderKuuga committed:
恢复通过MyBatisGenerator快速生成相关映射文件的代码

KamenRiderKuuga committed:
提交模拟提交题目并返回提交编号的爬虫示例代码

KamenRiderKuuga committed:
Merge branch 'dev' into dev-hanabi

gk008 committed:
Merge pull request #4 from qms-dev/dev-hanabi

增加字段，提交模拟提交题目的示例代码

KamenRiderKuuga committed:
引入mapstruct，用于entity和dto的映射

KamenRiderKuuga committed:
完成获取题目接口

KamenRiderKuuga committed:
解决https://github.com/qms-dev/qmoj-core/pull/5提到的问题

KamenRiderKuuga committed:
解决https://github.com/qms-dev/qmoj-core/pull/5提到的问题

KamenRiderKuuga committed:
获取所有题目摘要列表

KamenRiderKuuga committed:
统一后端题目标签字段名称为topicTags

KamenRiderKuuga committed:
统一后端题目标签字段名称为topicTags

KamenRiderKuuga committed:
今晚要重装系统，提交昨晚忘记的字段调整

KamenRiderKuuga committed:
整合Redis，使用Lettuce客户端

KamenRiderKuuga committed:
Merge branch 'dev' into dev-hanabi

KamenRiderKuuga committed:
实现提交题解以及查询判题结果功能

KamenRiderKuuga committed:
调整前端传入参数的分类，调整api路由规范

Comment from @21kl
https://github.com/qms-dev/qmoj-core/pull/5#discussion_r549094160
https://github.com/qms-dev/qmoj-core/pull/5#discussion_r549095110

Co-Authored-By: 21kl <45235187+21kl@users.noreply.github.com>

KamenRiderKuuga committed:
Merge pull request #5 from qms-dev/dev-hanabi

confirm merge:
引入mapstruct，用于entity和dto的映射
完成获取题目详情接口
实现获取所有题目摘要列表接口
整合Redis，使用Lettuce客户端
实现提交题解以及查询判题结果接口
